File: src/main/java/com/dozycoffee/wms/warehouse/application/port/in/GetLocationUseCase.java
package com.dozycoffee.wms.warehouse.application.port.in;

import com.dozycoffee.wms.warehouse.application.port.in.result.LocationResult;
import reactor.core.publisher.Mono;

public interface GetLocationUseCase {

    Mono<LocationResult> getById(Long locationId);
}

File: src/main/kotlin/com/dozycoffee/wms/warehouse/application/port/in/GetLocationUseCase.kt
package com.dozycoffee.wms.warehouse.application.port.`in`

import com.dozycoffee.wms.warehouse.application.port.`in`.result.LocationResult
import reactor.core.publisher.Mono

interface GetLocationUseCase {
    fun getById(locationId: Long): Mono<LocationResult>
}
